blobfuse/fileapis.cpp

```diff
@@ -491,6 +495,14 @@ int azs_truncate(const char * path, off_t off)
             return res;
         }
 
+        // If the file in cache has exactly the same size then there is no need to do trucate and upload
+        struct stat buf;
```

Comment:
How do we know that something has not changed just by the size of the file

Reply:
With truncate we can only set the size of a file to a certain number you can not change the data. So if the current size of file is same as what you are setting with truncate then there is no need to take any action. For e.g. if user create a new file of 100 bytes and then truncates it to 100, nothing changes. Just ignoring such calls because truncate for blobfuse results into uploading a new file with given size.

Reply:
If question is about write something to file and then truncate then anyway we will receive a flush call later which will upload, so that shall also be ignored here. This is what happens in case of SCP where after writing the data we receive a truncate call followed by a flush call.